### ISSUE TITLE
Run tests several times, using a combination of core/sxl versions

### ISSUE THREAD

**Issue by emiltin** [rsmp-nordic/rsmp_validator#406]
The core version is negotiated, so we should be able to run the test suite multiple times, using different core versions.

The sxl version is more tricky, since it's not negotiated, but simply stated by the site, see https://github.com/rsmp-nordic/rsmp_core/issues/65. Preferably, the core version would be negotiated too, so that the validator can try each sxl version in turn.

**Comment by emiltin** [comment]
@mariusrazvanplugaru, the latest main branch updates the M0022 tests, to correctly correctly cancel requests. there's also a new tests that checks that a request becomes stale if now cancelled.

Do you want to do a test run with the latest core and sxl?

**Comment by emiltin** [comment]
We should be able to use GitHub's matrix feature to run create a test job for each core version:

https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow

**Comment by mariusrazvanplugaru** [comment]
Hi, 
thanks for this. We are taking steps to do this test too. Let's think during the week. When we are ready we will certainly contact you.
KR,

**Comment by emiltin** [comment]
There should not be much to do on your side - the validator would run tests sevearl times, requesting different core versions during the handshake. We might need to modify the configs, since the core version would be set by the github action.

**Comment by emiltin** [comment]
Running on a matrix of core version is possible after #422, and is used for the gem tlc and supervisor tests.
Testing sites with differnt SXL version is not easy since we can't control the SXL from the supervisor (validator) side.

Closing, we can open a new issue specifically for how to test different site sxl versions.
